src/ray/common/client_connection.cc

```diff
@@ -261,8 +264,14 @@ void ClientConnection<T>::ProcessMessage(const boost::system::error_code &error)
   message_handler_(shared_ClientConnection_from_this(), read_type_, read_message_.data());
   int64_t interval = current_time_ms() - start_ms;
   if (interval > RayConfig::instance().handler_warning_timeout_ms()) {
-    RAY_LOG(WARNING) << "[" << debug_label_ << "]ProcessMessage with type " << read_type_
-                     << " took " << interval << " ms.";
+    const char *message_type;
+    if (message_type_enum_names_ == nullptr) {
+      message_type = std::to_string(read_type_).c_str();
```

Comment:
this might produce a memory leak? (i.e. the temporary string constructed by std::to_string goes out of scope and the pointer returned by c_str becomes invalid)

Reply:
Ah nice catch, thanks!